File: HackerRank/Problem-Solving/Implementation/DayOfProgrammer.py
#!/bin/python3
# AUTHOR :: RAHUL MISTRY
# DATE   :: 24 / 04 / 2023
import math
import os
import random
import re
import sys

#
# Complete the 'dayOfProgrammer' function below.
#
# The function is expected to return a STRING.
# The function accepts INTEGER year as parameter.
#

def dayOfProgrammer(year):
    # Write your code here
    if year==1918:
        return '29.09.1918'
    elif year <= 1917 and year % 4 == 0:
        return f'12.09.{year}'
    elif year % 4 == 0 and (year % 100 != 0 or year % 400 == 0):
        return f'12.09.{year}'
    else:
        return f'13.09.{year}'

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    year = int(input().strip())

    result = dayOfProgrammer(year)

    fptr.write(result + '\n')

    fptr.close()

File: HackerRank/Problem-Solving/Implementation/DayofProgrammer.cpp
██████    █████   ██   ██  ██    ██  ██             ███    ███  ██  ███████  ████████  ██████   ██    ██
██   ██  ██   ██  ██   ██  ██    ██  ██             ████  ████  ██  ██          ██     ██   ██   ██  ██ 
██████   ███████  ███████  ██    ██  ██             ██ ████ ██  ██  ███████     ██     ██████     ████  
██   ██  ██   ██  ██   ██  ██    ██  ██             ██  ██  ██  ██       ██     ██     ██   ██     ██   
██   ██  ██   ██  ██   ██   ██████   ███████        ██      ██  ██  ███████     ██     ██   ██     ██   


// AUTHOR :: RAHUL MISTRY
// DATE   :: 00 / 00 / 2023

#include <bits/stdc++.h>

using namespace std;

string ltrim(const string &);
string rtrim(const string &);

/*
 * Complete the 'dayOfProgrammer' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts INTEGER year as parameter.
 */

string dayOfProgrammer(int year) {
    string date;
    
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string year_temp;
    getline(cin, year_temp);

    int year = stoi(ltrim(rtrim(year_temp)));

    string result = dayOfProgrammer(year);

    fout << result << "\n";

    fout.close();

    return 0;
}

string ltrim(const string &str) {
    string s(str);

    s.erase(
        s.begin(),
        find_if(s.begin(), s.end(), not1(ptr_fun<int, int>(isspace)))
    );

    return s;
}

string rtrim(const string &str) {
    string s(str);

    s.erase(
        find_if(s.rbegin(), s.rend(), not1(ptr_fun<int, int>(isspace))).base(),
        s.end()
    );

    return s;
}
